File: src/Frontend/src/app/dashboard/Models/variant-inventory-status.model.ts
import { PurchaseInvoiceDetailsModel } from "./purchase-invoice-details.model";

export class VariantInventoryStatusModel {
    productId: number = 0;
    variantId: number = 0;
    productName: string = '';
    variantName: string = '';
    sku: string = '';
    lots: Array<PurchaseInvoiceDetailsModel> = [];
}

File: src/Backend/Inventory.Data/Models/Product/VariantInventoryStatusModel.cs
using Inventory.Data.Models.Purchase;

namespace Inventory.Data.Models.Product
{
    public class VariantInventoryStatusModel
    {
        public int ProductId { get; set; }
        public int VariantId { get; set; }
        public string ProductName { get; set; }
        public string VariantName { get; set; }
        public string ProductCategory { get; set; }
        public string ProductSubCategory { get; set; }
        public string ProductImage { get; set; }
        public string SKU { get; set; }
        public IList<PurchaseInvoiceDetailsModel> Lots { get; set; }
    }
}
